app/Jobs/AdsSendOneSwash.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * Copyright (c) 2018-2021 Adshares sp. z o.o.
+ *
+ * This file is part of AdServer
+ *
+ * AdServer is free software: you can redistribute and/or modify it
+ * under the terms of the GNU General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * AdServer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AdServer. If not, see <https://www.gnu.org/licenses/>
+ */
+
+namespace Adshares\Adserver\Jobs;
+
+use Adshares\Ads\AdsClient;
+use Adshares\Ads\Command\SendOneCommand;
+use Adshares\Ads\Driver\CommandError;
+use Adshares\Ads\Exception\CommandException;
+use Adshares\Ads\Util\AdsValidator;
+use Adshares\Adserver\Exceptions\JobException;
+use Adshares\Adserver\Models\UserLedgerEntry;
+use Exception;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Log;
+
+class AdsSendOneSwash implements ShouldQueue
+{
+    use Dispatchable;
+    use Queueable;
+    use SerializesModels;
+
+    private const QUEUE_NAME = 'ads';
+
+    private const QUEUE_TRY_AGAIN_EXCEPTION_CODES = [
+        CommandError::LOW_BALANCE,
+        CommandError::USER_BAD_TARGET,
+        CommandError::LOCK_USER_FAILED,
+    ];
+
+    /**
+     * @var string recipient address
+     */
+    private $addressTo;
+
+    /**
+     * @var int transfer amount
+     */
+    private $amount;
+
+    /**
+     * @var null|string optional message
+     */
+    private $message;
+
+    /**
+     * @var string batchId
+     */
+    private $batchId;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param string $batchId
+     * @param string $addressTo
+     * @param int $amount
+     * @param null|string $message
+     */
+    public function __construct(string $batchId, string $addressTo, int $amount, ?string $message = null)
+    {
+        $this->batchId = $batchId;
+        $this->addressTo = $addressTo;
+        $this->amount = $amount;
+        $this->message = $message;
+        $this->queue = self::QUEUE_NAME;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @param AdsClient $adsClient
+     *
+     * @throws JobException
+     */
+    public function handle(AdsClient $adsClient): void
+    {
+        $userLedger = UserLedgerEntry::getFirstRecordByBatchId($this->batchId);
```

Comment:
If this line fails for any reason, the job will enter a loop. set a max attempt.